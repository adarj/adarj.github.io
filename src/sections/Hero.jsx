import React from 'react';
import MainHeading from '../components/MainHeading';
import heroBg from '../images/hero-bg.jpg';

const title = "AKIL DARJEAN";
const subtitle = "SOFTWARE DEVELOPER";
const links = {
  email: "mailto:akildarjean@protonmail.com",
  github: "https://github.com/adarj",
  linkedin: "https://linkedin.com/in/adarj"
};

const Hero = () => (
  <MainHeading image={heroBg}
               title={title}
               subtitle={subtitle}
               {...links} />
);

export default Hero;
