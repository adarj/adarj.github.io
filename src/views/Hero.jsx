import React from "react"
import LazyHero from "react-lazy-hero"
import heroBg from "../images/hero-bg.jpg"

const Hero = ({ children }) => (
  <LazyHero imageSrc={heroBg} color="#000" opacity={0.6} minHeight="100vh">
    {children}
  </LazyHero>
)

export default Hero
