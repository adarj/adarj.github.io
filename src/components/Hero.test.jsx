import React from 'react';
import { shallow } from 'enzyme';

import Hero from './Hero';
import heroBg from '../images/hero-bg.jpg';

const text = {
  title: "John Doe",
  subtitle: "Lorem Ipsum"
};

const links = {
  email: "mailto:john.doe@mail.com",
  github: "https://github.com",
  linkedin: "https://linkedin.com"
};

describe('Hero', () => {
  it('renders correctly', () => {
    const tree = shallow(<Hero image={heroBg} {...text} {...links} />);
    expect(tree).toMatchSnapshot();
  });
});

