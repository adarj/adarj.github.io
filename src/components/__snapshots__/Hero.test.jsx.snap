// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<LazyHero
  color="#000"
  imageSrc="test-file-stub"
  isCentered={true}
  isFixed={false}
  minHeight="70vh"
  opacity={0.6}
  parallaxOffset={0}
  transitionDuration={600}
  transitionTimingFunction="ease-in-out"
>
  <styled.div>
    <styled.h1
      data-cy="title"
    >
      John Doe
    </styled.h1>
    <styled.h2
      data-cy="subtitle"
    >
      Lorem Ipsum
    </styled.h2>
  </styled.div>
  <styled.div>
    <ContextProvider
      value={
        Object {
          "color": "white",
          "size": "1.50em",
        }
      }
    >
      <styled.a
        data-cy="email-link"
        href="mailto:john.doe@mail.com"
        target="_blank"
      >
        <FaEnvelope />
      </styled.a>
      <styled.a
        data-cy="github-link"
        href="https://github.com"
        target="_blank"
      >
        <FaGithub />
      </styled.a>
      <styled.a
        data-cy="linkedin-link"
        href="https://linkedin.com"
        target="_blank"
      >
        <FaLinkedin />
      </styled.a>
    </ContextProvider>
  </styled.div>
</LazyHero>
`;
